Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs

```diff
@@ -155,11 +156,22 @@ await container.CreateItemAsync(value, new PartitionKey(value.PartitionKey), can
         /// <inheritdoc/>
         public async ValueTask<TItem> UpdateAsync(
             TItem value,
-            CancellationToken cancellationToken = default)
+            CancellationToken cancellationToken = default,
+            bool verifyEtag = false)
         {
             (bool optimizeBandwidth, ItemRequestOptions options) = RequestOptions;
             Container container = await _containerProvider.GetContainerAsync().ConfigureAwait(false);
 
+            if (verifyEtag)
+            {
+                if (value is not IItemWithEtag valueWithEtag)
+                {
+                    throw new InvalidEtagConfigurationException($"{typeof(TItem).Name} is not assignable to {nameof(IItemWithEtag)}. You cannot specify {nameof(verifyEtag)} is true and not provide a value without an implementation of {nameof(IItemWithEtag)}.");
+                }
+
+                options.IfMatchEtag = valueWithEtag.Etag;
+            }

```

Comment:
```suggestion
            if (verifyEtag && value is not IItemWithEtag valueWithEtag)
            {
                throw new InvalidEtagConfigurationException(
                    $"{typeof(TItem).Name} is not assignable to {nameof(IItemWithEtag)}. " +
                    $"You cannot specify {nameof(verifyEtag)} is true and not provide a " +
                    $"value without an implementation of {nameof(IItemWithEtag)}.");

                options.IfMatchEtag = valueWithEtag.Etag;
            }
```

Minimize nesting of `if` by using `&&`.